src/wechaty_puppet/file_box/file_box.py

```diff
@@ -92,9 +93,40 @@ def type(self) -> FileBoxType:
         """get filebox type"""
         return FileBoxType(self.boxType)
 
-    def sync_remote_name(self):
-        """sync remote name"""
-        pass
+    async def ready(self):
+        """
+        sync the name from remote
+        """
+        if self.type() == FileBoxType.Url:
+            await self.sync_remote_name()
+
+    async def sync_remote_name(self):
+        """sync remote name
+        refer : https://developer.mozilla.org/en-US/docs/Web/HTTP/
+                Headers/Content-Disposition
+
+                Content-Disposition: attachment; filename="cool.html"
+
+        # wujingjing comment 2020-6-29:  according the implementation of
+            file-box: https://github.com/huan/file-box/blob/e42d7207bb1cf5
+                b76afb8ead6f72715f4a197b35/src/misc.ts#L66
+
+            headers in requests package doesn't contains attribute:
+                content-disposition, so we need to change the way to sync
+                file-name from url type
+
+            I find the better way to extract file name from url:
+                https://stackoverflow.com/questions/18727347/how-to-extract-a-
+                filename-from-a-url-append-a-word-to-it/18727481#18727481
+        """
+        if self.boxType is not FileBoxType.Url:
```

Comment:
`self.boxTyle` is a number, which can not be compared to FileBoxType.Url

Reply:
Yes, it's my fault to write this type checking code.  I think the correct type checking is :
```python
file_box_type = self.type()
if file_box_type != FileBoxType.Url:
    raise TypeError('type <{0}> is not remote'.format(
        file_box_type.name))
```
Glad to see that you can review  so detailed checking, please go ahead to all pr.